src/libstd/io/stdio.rs

```diff
@@ -872,6 +872,84 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Locks the current handle and reads a line of input, returning a `String` containing
+/// the input.
+///
+/// If you need more explicit control over
+/// locking, see the [`Stdin::lock`] and [`Stdout::lock`] methods.
+///
+/// [`Stdin::lock`]: struct.Stdin.html#method.lock
+/// [`Stdout::lock`]: struct.Stdout.html#method.lock
+///
+/// ### Note: Safety and Usage
+/// This function is not the recommended solution to get user input, it should
+/// only be used for simple actions. If you require more explicit control over
+/// getting user input, see the [`Stdin::read_line`] method.
+/// 
+/// [`Stdin::read_line`]: struct.Stdin.html#method.read_line
+///
+/// # Examples
+///
+/// ```rust
+/// use std::io::inputln;
+///
+/// fn main() {
+///     let user_input = inputln("Please enter some text: ");
+///         
+///     println!("You typed: {}", user_input);
+///         
+/// }
+/// ```
+pub fn inputln(prompt: &str) -> String {
+    stdout()
+        .write(prompt.as_bytes())
+        .unwrap();
+    let mut input = String::new();
+    let _ = stdout().flush();
+    stdin()
+        .read_line(&mut input)
+        .expect("Failed to read from stdin");
+    if ['\n', '\r'].contains(&input.chars().next_back().unwrap()) {
```

Comment:
This removes either an `\r` or `\n` char.

This differs from the example you give in the PR description which removes either `\n` or `\r` or `\r\n`:
```rust
if let Some('\n')=s.chars().next_back() {
    s.pop();
}
if let Some('\r')=s.chars().next_back() {
    s.pop();
}
```

Reply:
The function now accounts for all possible options.